plugins/tff_backend/rivine.py

```diff
@@ -0,0 +1,246 @@
+    url = '%s/blocks/%s' % (get_explorer_base_url(), height)
+    r = do_get_request(url)
+    return r['block']
+
+
+def get_info_by_hash(hash_):
+    url = '%s/hashes/%s' % (get_explorer_base_url(), hash_)
+    response = urlfetch.fetch(url, deadline=10)  # type: urlfetch._URLFetchResult
+    try:
+        content = json.loads(response.content)
+    except:
+        content = response.content
+    if response.status_code == 200:
+        return content
+    elif response.status_code == 400 and 'unrecognized hash' in content['message']:
+        return None
+    else:
+        err = HttpException(content['message'] if isinstance(content, dict) else content)
+        err.http_code = response.status_code
+        raise err
+
+
+def get_transactions(address, status=None):
+    if status and status not in (TransactionStatus.UNCONFIRMED, TransactionStatus.CONFIRMED):
+        return []
+    info = get_info_by_hash(address)
+    if not info or info['hashtype'] != 'unlockhash':
+        return []
+
+    live_block_height = get_block_height() - MATURITY_DEPTH
+    transactions = []
+    for t in info['transactions']:
+        if t['height'] > live_block_height:
+            t_status = TransactionStatus.UNCONFIRMED
+        else:
+            t_status = TransactionStatus.CONFIRMED
+
+        if status and status == TransactionStatus.CONFIRMED and t_status != TransactionStatus.CONFIRMED:
+            continue
+        elif status and status == TransactionStatus.UNCONFIRMED and t_status != TransactionStatus.UNCONFIRMED:
+            continue
+        if not t['coinoutputids'] or len(t['coinoutputids']) == 0:
+            continue
+
+        for sco_id, co in zip(t['coinoutputids'], t['rawtransaction']['data']['coinoutputs']):
+            # todo investigate if correct when fully spent
+            if co['unlockhash'] != address:
+                continue
+
+            transactions.append({'id': t['id'],
+                                 'height': t['height'],
+                                 'timestamp': get_block_by_height(t['height'])['rawblock']['timestamp'],
+                                 'output_id': sco_id,
+                                 'inputs': t['coininputoutputs'],
+                                 'outputs': t['rawtransaction']['data']['coinoutputs'],
+                                 'amount': unicode(co['value']),
+                                 'currency': TOKEN_TFT,
+                                 'status': t_status,
+                                 'spent': False})
+
+    if not transactions:
+        return []
+
+    for t in info['transactions']:
+        if not t['rawtransaction']['data']['coininputs'] or len(t['rawtransaction']['data']['coininputs']) == 0:
+            continue
+
+        for ci in t['rawtransaction']['data']['coininputs']:
+            for t in transactions:
+                if ci['parentid'] != t['output_id']:
+                    continue
+                t['spent'] = True
```

Comment:
Here `t` shouldn't be modified, because it belongs to the `info['transactions']` list, which isn't used anymore in this method.

You should get the correct transaction from the `transactions` list (maybe best to modify to a dict) and modify spent in that object.